### PR TITLE
feat(topics/advanced): Add the storage backend section

### LINKED ISSUE #422: Add storage backend section
The install guide is missing the section on storage backend configuration.

### PR DESCRIPTION
**Pull request by hickeyma**
Closes #422

### CODE REVIEW

**Comment by hickeyma** [comment]
@emaincourt Thank you for the suggestion on adding the note. I have updated the doc with the change.

File: content/en/docs/topics/advanced.md [hunk 1]
```diff
@@ -143,3 +143,68 @@ func main() {
...
+```shell
+kubectl get secret --all-namespaces -l "owner=helm"
+```
+
+**PRODUCTION NOTES**: It is not recommended to use ConfigMaps for sensitive data as
+it is stored unencoded. Secrets data is stored base64-encoded.
+
+### SQL storage backend
+
+There is a beta SQL storage backend that stores release information in an SQL database.
```
Comment (mattfarina) [inline content/en/docs/topics/advanced.md]: The [pull request for the feature](https://github.com/helm/helm/pull/7635) does not note it as "beta". Do we have more than supported or experimental? Should this be noted as "beta" if we don't really have that designation?

File: content/en/docs/topics/advanced.md [hunk 2]
```diff
@@ -143,3 +143,68 @@ func main() {
...
+If you want to switch from the default backend to the ConfigMap
+backend, you'll have to do the migration for this on your own. You can retrieve
+release information with the following command:
+
+```shell
+kubectl get secret --all-namespaces -l "owner=helm"
+```
+
+**PRODUCTION NOTES**: It is not recommended to use ConfigMaps for sensitive data as
+it is stored unencoded. Secrets data is stored base64-encoded.
```
Comment (mattfarina) [inline content/en/docs/topics/advanced.md]: Anyone who cares about sensitive information storage is going to see the "base64-encoded" and know that's not secure. We don't want them to think we think it's secure. I would add something to the end like...

> ... and secrets can be configured for encrypted storage.

The last part I'd link to https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/

File: content/en/docs/topics/advanced.md [hunk 3]
```diff
@@ -143,3 +143,68 @@ func main() {
...
+export HELM_DRIVER_SQL_CONNECTION_STRING=postgresql://helm-postgres:5432/helm?user=helm&password=changeme
+```
+
+> Note: Only Postgres is supported at this moment. This means that `HELM_DRIVER_SQL_DIALECT`
+needs to be set to `postgres`.
+
+**PRODUCTION NOTES**: It is recommended to:
+- Change the default username and password of the SQL database
+- Enable SSL to tighten security access
+- Enable [permission management](/docs/permissions_sql_storage_backend/) to mirror Kubernetes
```
Comment (mattfarina) [inline content/en/docs/topics/advanced.md]: I think this is a bad link. What's the right place for the permission management link?

File: content/en/docs/topics/advanced.md [hunk 4]
```diff
@@ -143,3 +143,68 @@ func main() {
...
+```shell
+export HELM_DRIVER=sql
+export HELM_DRIVER_SQL_DIALECT=postgres
+export HELM_DRIVER_SQL_CONNECTION_STRING=postgresql://helm-postgres:5432/helm?user=helm&password=changeme
+```
+
+> Note: Only Postgres is supported at this moment. This means that `HELM_DRIVER_SQL_DIALECT`
+needs to be set to `postgres`.
+
+**PRODUCTION NOTES**: It is recommended to:
```
Comment (mattfarina) [inline content/en/docs/topics/advanced.md]: Do we want to tell them how to setup their database for production? Is there more general advice we should give them and then provide a link?

File: content/en/docs/topics/advanced.md (hunk 2 above)
Comment (hickeyma) [reply content/en/docs/topics/advanced.md]: Thats a good point. Added the clarification.

**Comment by hickeyma** [comment]
Thanks for the reviews @mattfarina @emaincourt

**Pull request merged**